processor/src/main/java/com/dropbox/kaiken/processor/InjectableActivityValidator.kt

```diff
@@ -1,55 +1,53 @@
 package com.dropbox.kaiken.processor
 
-import com.dropbox.kaiken.processor.internal.error
-import com.dropbox.kaiken.processor.internal.implementsInjectionHolder
-import com.dropbox.kaiken.processor.internal.isAbstract
-import com.dropbox.kaiken.processor.internal.isAndroidActivity
-import com.dropbox.kaiken.processor.internal.isPublic
-import javax.annotation.processing.Messager
-import javax.tools.Diagnostic
-
-internal class InjectableActivityValidator(
-    private val messager: Messager
-) {
-    fun isValid(annotatedActivity: InjectableAnnotatedActivity): Boolean {
-
-        val typeElement = annotatedActivity.annotatedActivityElement
-
-        return when {
-            !typeElement.isPublic() -> {
-                messager.error(typeElement, "The class ${typeElement.qualifiedName} is not public")
-                false
-            }
-
-            typeElement.isAbstract() -> {
-                messager.error(typeElement, "The class ${typeElement.qualifiedName} is abstract")
-                false
-            }
-
-            !typeElement.isAndroidActivity() -> {
-                messager.error(
-                    typeElement,
-                    "The class ${typeElement.qualifiedName} is not an Android activity"
-                )
-                messager.printMessage(Diagnostic.Kind.NOTE, typeElement.superclass.kind.name)
-                false
-            }
-
-            !typeElement.implementsInjectionHolder() -> {
-                messager.error(
-                    typeElement,
-                    "The class ${typeElement.qualifiedName} does not implement" +
-                        " DependencyProviderResolver"
-                )
-                typeElement.interfaces.forEach {
-                    messager.printMessage(Diagnostic.Kind.NOTE, it.kind.name)
-                }
-                false
-            }
-
-            else -> {
-                true
-            }
-        }
+import org.jetbrains.kotlin.descriptors.ClassDescriptor
+import org.jetbrains.kotlin.descriptors.ClassifierDescriptor
+import org.jetbrains.kotlin.descriptors.DescriptorVisibilities
+import org.jetbrains.kotlin.name.FqName
+import org.jetbrains.kotlin.psi.KtClassOrObject
+import org.jetbrains.kotlin.resolve.DescriptorUtils
+import org.jetbrains.kotlin.resolve.descriptorUtil.fqNameSafe
+import org.jetbrains.kotlin.resolve.descriptorUtil.getAllSuperClassifiers
+
+internal fun validateActivity(descriptor: ClassDescriptor?, clazz: KtClassOrObject) {
+    check(descriptor != null) {
+        "The ClassDescriptor for ${clazz.name} is null"
+    }
+
+    check(clazz.fqName != null) {
+        "The ClassDescriptor for ${clazz.name} is null"
+    }
+
+    val fqName: FqName = requireNotNull(clazz.fqName) {
```

Comment:
I'm not sure we need the check then a require not null. That should be same result

Reply:
Done